packages/create-project/src/index.ts

```diff
@@ -1,151 +1,181 @@
+    process.exit(1);
   }
-  throw new TypeError(`Invalid sample name ${sampleName}`);
-}
 
-async function createProject(projectPath: string, useYarn: boolean, temporalVersion: string, sample: string) {
-  const root = path.resolve(projectPath);
-  const src = path.resolve(root, 'src');
-  const name = path.basename(root);
-  await mkdir(root);
-  const packageJson = { ...packageJsonBase, name };
-  await writePrettyJson(path.join(root, 'package.json'), packageJson);
-  await mkdir(src);
-  await mkdir(path.join(src, 'interfaces'));
-  await mkdir(path.join(src, 'workflows'));
-  await writePrettyJson(path.join(root, 'tsconfig.json'), tsConfig);
-  const sampleDir = path.join(__dirname, '../samples');
-  const template = getTemplate(sample);
-  await template.copySources(sampleDir, src);
-  if (useYarn) {
-    await spawn('yarn', ['install'], { cwd: root, stdio: 'inherit' });
-    await spawn('yarn', ['add', `temporalio@${temporalVersion}`], { cwd: root, stdio: 'inherit' });
-  } else {
-    await spawn(npm, ['install'], { cwd: root, stdio: 'inherit' });
-    await spawn(npm, ['install', `temporalio@${temporalVersion}`], { cwd: root, stdio: 'inherit' });
+  const resolvedProjectPath = path.resolve(projectPath);
+  const projectName = path.basename(resolvedProjectPath);
+
+  const { valid, problems } = validateNpmName(projectName);
+  if (!valid) {
+    console.error(
+      `Could not create a project called ${chalk.red(`"${projectName}"`)} because of npm naming restrictions:`
+    );
+
+    problems?.forEach((p) => console.error(`    ${chalk.red.bold('*')} ${p}`));
+    process.exit(1);
   }
-}
 
-async function init() {
-  const { _: args, ...opts } = arg({
-    '--use-yarn': Boolean,
-    '--temporal-version': String,
-    '--sample': String,
-  });
-  if (args.length !== 1) {
-    throw new UsageError();
+  let example = program.example;
+
+  // `example` is true when --example is used by itself
+  if (typeof example !== 'string') {
+    const res = await prompts({
+      type: 'text',
+      name: 'example',
+      message: 'Which template would you like to use?',
+      initial: 'hello-world',
+      // validate: (name) => {
+      //   const validation = validateNpmName(path.basename(path.resolve(name)));
+      //   if (validation.valid) {
+      //     return true;
+      //   }
+      //   return 'Invalid project name: ' + validation.problems![0];
+      // },
+    });
+
+    if (typeof res.example === 'string') {
+      example = res.example;
+    }
+  }
+
+  if (typeof example !== 'string') {
+    console.log();
+    console.log('Please specify which example:');
+    console.log(`  ${chalk.cyan(program.name())} --example ${chalk.green('[name]|[github-url]')}`);
+    console.log();
+    console.log('For example:');
+    console.log(`  ${chalk.cyan(program.name())} --example ${chalk.green('hello-world')}`);
+    console.log();
+    console.log(`Run ${chalk.cyan(`${program.name()} --help`)} to see all options.`);
```

Comment:
Should we use `console.error`? It's common practice in CLIs to log errors to stderr.